modules/reindex/src/main/java/org/elasticsearch/index/reindex/AbstractBaseReindexRestHandler.java

```diff
@@ -61,7 +76,11 @@ protected AbstractBaseReindexRestHandler(Settings settings, Client client,
     }
 
     protected void execute(RestRequest request, Request internalRequest, RestChannel channel) throws IOException {
-        internalRequest.setRequestsPerSecond(request.paramAsFloat("requests_per_second", internalRequest.getRequestsPerSecond()));
+        Float requestsPerSecond = parseRequestsPerSecond(request);
+        if (requestsPerSecond != null) {
```

Comment:
Totally up to you (whether you want to or not), but maybe `setRequestsPerSecond` should handle the null case so you don't have to worry about checking it post-parse? What do you think? (I'm fine either way)